File: automation/jsr223/python/personal/scenes.py
from core.actions import ScriptExecution
from core.rules import rule
from core.triggers import when
from core.actions import LogAction
from java.time import ZonedDateTime as DateTime


tsceneStartup = None


@rule("StartUp - set up Item Scene_Goodnight", description="StartUp - set up Item Scene_Goodnight", tags=["Scene"])
@when("System started")
def scene_Goodnight_init(event):
    LogAction.logInfo("StartUp - set up Item Scene_Goodnight", "StartUp - set up Item Scene_Goodnight")
    # events.postUpdate("BridgeLightSensorState", "OFF")
    global tsceneStartup
    if tsceneStartup is None:
        tsceneStartup = ScriptExecution.createTimer(DateTime.now().plusSeconds(45), lambda: events.postUpdate("Scene_Goodnight", "OFF"))


tgoodnight = None


@rule("Goodnight Going to bed", description="Goodnight Going to bed", tags=["scene"])
@when("Item Scene_Goodnight changed from OFF to ON")
def Scene_Goodnight(event):
    # Scene_Goodnight.log.info("::Boiler_Control rule -> A Heater recieved a command - updating boiler state::")
    LogAction.logError("Scene_Goodnight", "goodnight going to bed")
    global tgoodnight
    global CT_HPSP_Night

    events.sendCommand("CT_FairyLights433Socket", "OFF")
    events.sendCommand("ZbColourBulb01Switch", "OFF")
    events.sendCommand("ZbColourBulb02Switch", "OFF")

    events.sendCommand("radio", "OFF")
    events.sendCommand("vCT_TVKodiSpeakers", "OFF")
    events.postUpdate("CT_Heating_PresetTempNormal", items["CT_HPSP_Night"].toString())
    # events.sendCommand("CT_Heating_PresetTempNormal", items["CT_HPSP_Night"].toString())
    events.postUpdate("FR_Heating_PresetTempNormal", items["FR_HPSP_Night"].toString())
    events.postUpdate("ER_Heating_PresetTempNormal", items["ER_HPSP_Night"].toString())
    events.postUpdate("AT_Heating_PresetTempNormal", items["AT_HPSP_Night"].toString())
    events.postUpdate("BR_Heating_PresetTempNormal", items["BR_HPSP_Night"].toString())
    events.postUpdate("OF_Heating_PresetTempNormal", items["OF_HPSP_Night"].toString())
    events.postUpdate("HL_Heating_PresetTempNormal", items["HL_HPSP_Night"].toString())
    events.sendCommand("Heating_UpdateHeaters", "OFF")
    events.sendCommand("Heating_UpdateHeaters", "ON")
    events.postUpdate("Scene_Goodnight", "OFF")
    events.sendCommand("workLightsPowerSocket", "OFF")
    events.sendCommand("workLightsPowerSocket", "OFF")
    # events.sendCommand("Heating_UpdateHeaters", "ON") #trigger updating of heaters and boiler etc

    tgoodnight = ScriptExecution.createTimer(DateTime.now().plusSeconds(300), lambda: events.sendCommand("ZbWhiteBulb01Switch", "OFF"))

File: automation/js/scenes.js
const {
  log, items, rules, actions, triggers,
} = require('openhab');
const { myutils } = require('personal');

const logger = log('scenes');
const { timeUtils } = require('openhab_rules_tools');

let tsceneStartup;
const resetSceneGoodnightDelay = 45 * 1000;

scriptLoaded = function () {
  logger.info('StartUp - set up Item Scene_Goodnight');
  if (tsceneStartup === undefined) {
    tsceneStartup = actions.ScriptExecution.createTimer(time.toZDT((resetSceneGoodnightDelay)), () => {
      items.getItem('Scene_Goodnight').postUpdate('OFF');
    });
  }
};

let tgoodnight;

rules.JSRule({
  name: 'Goodnight Going to bed',
  description: 'Goodnight Going to bed',
  triggers: [triggers.ItemStateChangeTrigger('Scene_Goodnight', 'OFF', 'ON')],
  execute: (event) => {
    logger.info(`Goodnight Going to bed, triggering item name: ${event.itemName} : ,  received  update event.receivedState: ${event.receivedState}`);

    items.getItem('CT_FairyLights433Socket').sendCommand('OFF');

    items.getItem('ZbColourBulb01_switch').sendCommand('OFF');
    items.getItem('ZbColourBulb02_switch').sendCommand('OFF');

    items.getItem('radio').sendCommand('OFF');
    items.getItem('gCT_TVKodiSpeakers').sendCommand('OFF');
    // items.getItem('vCT_TVKodiSpeakers2').sendCommand('OFF');

    // turn off all heating

    // items.getItem('Heating_UpdateHeaters').sendCommand('OFF');
    // items.getItem('Heating_UpdateHeaters').sendCommand('ON');

    items.getItem('gLightCyclers').postUpdate('OFF');

    items.getItem('Scene_Goodnight').postUpdate('OFF');
    items.getItem('workLightsPowerSocket').sendCommand('OFF');

    tgoodnight = actions.ScriptExecution.createTimer(time.toZDT((300 * 1000)), () => {
      items.getItem('ZbWhiteBulb01Switch').sendCommand('OFF');
    });
  },
});
